apis/cloudflare/src/realtime-logger.ts

```diff
@@ -0,0 +1,772 @@
+import * as Braintrust from "braintrust/browser";
+import { z } from "zod";
+import { makeWavFile, makeMp3File } from "@braintrust/proxy/utils";
+import { PcmAudioFormat, ProxyLoggingParam } from "@braintrust/proxy/schema";
+
+// Type definitions adapted from:
+// https://github.com/openai/openai-realtime-api-beta/blob/0126e4bfc19901598c3f20d0a4b32bb3e0bea376/lib/client.js
+// https://platform.openai.com/docs/api-reference/realtime-client-events
+// Includes some modifications
+// - Where the OpenAI implementation differs from their type spec.
+// - Replace the fields we don't use with `z.any()` for more permissive parsing.
+const baseMessageSchema = z.object({
+  event_id: z.string(),
+});
+
+const audioFormatTypeSchema = z.enum(["pcm16", "g711_ulaw", "g711_alaw"]);
+type AudioFormatType = z.infer<typeof audioFormatTypeSchema>;
+
+const turnDetectionServerVadTypeSchema = z.object({
```

Comment:
small nit: split the schema definitions into a separate file

Reply:
Done